ui/de-lib/src/main/java/org/iplantc/de/analysis/client/presenter/AnalysesPresenterImpl.java

```diff
@@ -353,8 +351,13 @@ public void onShareSelected(List<Analysis> selected) {
 
     @Override
     public void setCurrentFilter(AnalysisFilter filter) {
-       this.currentFilter = filter;
-       loadAnalyses(currentFilter);
+        if(filter == null) {
+            currentFilter =filter;
+            return;
```

Comment:
LoadAnalyses method has been refactored to construct the query that should be sent to service while currentFilter is just a filed that track what UI is displaying in the drop down (later used for session restore). When the drop down value changes, i am updating the currentFilter value  and calling loadAnalyses to update the view. Not sure if this answers your questions but thats my thinking here...